apps/system-e2e/src/tests/service-portal/smoke/access-control.spec.ts

```diff
@@ -0,0 +1,107 @@
+import { BrowserContext, expect, test } from '@playwright/test'
+import { urls } from '../../../support/utils'
+import { session } from '../../../support/session'
+import { helpers } from '../../../support/locator-helpers'
+
+test.use({ baseURL: urls.islandisBaseUrl })
+
+test.describe('Service portal, access control', () => {
+  let context: BrowserContext
+
+  test.beforeAll(async ({ browser }) => {
+    context = await session({
+      browser: browser,
+      storageState: 'service-portal-faereyjar.json',
+      homeUrl: `${urls.islandisBaseUrl}/minarsidur`,
+      phoneNumber: '0102399',
+      idsLoginOn: true,
+    })
+  })
+
+  test.afterAll(async () => {
+    await context.close()
+  })
+
+  test('can remove delegations', async () => {
+    // Arrange
+    const page = await context.newPage()
+    await page.goto('/minarsidur/adgangsstyring/umbod')
+    await expect(
+      page.locator(
+        '[data-testid="access-card"], [data-testid="delegations-empty-state"]',
+      ),
+    ).not.toHaveCount(0)
+
+    // Act
+    const delegations = page
+      .locator('data-testid=access-card')
+      .filter({ hasText: 'Gervimaður Ameríku' })
+    const deleteButton = delegations.first().locator('role=button[name=Eyða]')
+    const confirmButton = page.locator(
+      '[data-dialog-ref="access-delete-modal"] >> role=button[name*=Eyða]',
+    )
+    const count = await delegations.count()
+
+    for (let i = 0; i < count; ++i) {
+      await deleteButton.click()
+      await confirmButton.click()
+      await confirmButton.waitFor({ state: 'hidden' })
+    }
+
+    // Assert
+    expect(delegations).toHaveCount(0)
+  })
+
+  test('can create delegation', async () => {
+    // Arrange
+    const page = await context.newPage()
+    await page.goto('/minarsidur/adgangsstyring/umbod')
+
+    // Act
+    await page.locator('role=button[name="Nýtt umboð"]').click()
+    await page.locator('role=textbox[name*="Kennitala"]').click()
+    // eslint-disable-next-line local-rules/disallow-kennitalas
+    await page.locator('role=textbox[name*="Kennitala"]').fill('010130-2989')
+
+    await page.locator('data-testid=proceed').click()
+
+    const access = page.locator('input[type=checkbox][id*=scope]')
+    await expect(access).not.toHaveCount(0)
+    const accessCount = await access.count()
+    for (let i = 0; i < accessCount; i++) {
+      await access.nth(i).setChecked(true)
+    }
+    await page.locator('data-testid=proceed').click()
+    await page.locator('role=button[name=Staðfesta]').click()
```

Comment:
Check out https://playwright.dev/docs/api/class-locator#locator-evaluate-all

Reply:
I admit `Locator#evaluateAll` might be a bit cleaner here, but I like staying in `Locator` land in Playwright tests.